tests/unit/test_settings_updates.py

```
@@ -434,3 +434,235 @@ def make_request():
+        assert len(calls) == 1
+
+    def test_run_real_update_rejects_bad_script_path(self, monkeypatch):
+        """Defense-in-depth: _run_real_update must also validate its inputs."""
+        import pytest
+
+        import blueprints.settings as mod
+
+        popen_mock = MagicMock()
+        monkeypatch.setattr(mod.subprocess, "Popen", popen_mock)
+        with pytest.raises(ValueError, match="Invalid update script path"):
+            mod._run_real_update("/tmp/evil.sh", target_tag="v1.0.0")
```

Ruff (0.15.9)

[error] Probable insecure usage of temporary file or directory: "/tmp/evil.sh"

(S108)